cmd/testwrapper/testwrapper.go

```diff
@@ -28,18 +28,21 @@ import (
 
 const maxAttempts = 3
 
+// testAttempt keeps track of the test name, outcome, logs, and if the test is flakey.
+// After running the tests, each testAttempt is written to a json file.
 type testAttempt struct {
-	name          testName
-	outcome       string // "pass", "fail", "skip"
+	Name          testName `json:"name,omitempty,inline"`
+	Outcome       string   `json:"outcome,omitempty"` // "pass", "fail", "skip"
 	logs          bytes.Buffer
-	isMarkedFlaky bool // set if the test is marked as flaky
-
-	pkgFinished bool
+	IsMarkedFlaky bool `json:"is_marked_flaky,omitempty"` // set if the test is marked as flaky
+	AttemptCount  int  `json:"attempt_count,omitempty"`
+	PkgFinished   bool `json:"pkg_finished,omitempty"`
```

Comment:
don't use `_` in names. Actually don't even override the names. Just add `,omitempty`